File: model-lib/src/main/java/view/ConstructView.java
package view;

import lombok.*;

import java.awt.*;
import java.util.UUID;

@Builder
@Getter
@RequiredArgsConstructor
@Setter
public class ConstructView {
    @NonNull
    private UUID associatedConstructId;
    private Color backColor;
    private String content;
    @NonNull
    private UUID id;
    private Font font;
    private Shape shape;
    private Stroke stroke;
    private Color strokeColor;
}

File: model-lib/src/main/kotlin/ru/psu/view/ConstructView.kt
package ru.psu.view

import java.awt.Color
import java.awt.Font
import java.awt.Shape
import java.awt.Stroke
import java.util.*

//Класс, соответствующий графическому представлению одной отдельно взятой конструкции
class ConstructView(
        val id: UUID, //Id графического представления
        var associatedConstructId:UUID //Id конструкции, которой соответствует отображение
) {
    var backColor:Color = Color.WHITE //Основной цвет граф. представления
    var content:String = "" //Текстовая подпись
    var font:Font? = null //Текстовая подпись
    var shape:Shape? = null //Контур графического представления
    var stroke: Stroke? = null //Тип границ контура (сплошные / прерывистые)
    var strokeColor:Color = Color.BLACK //Цвет контура графического представления
}